test/comparison-tests/projectReferencesNoSourceMap/lib/tsconfig.json

```diff
@@ -0,0 +1,8 @@
+{
+  "compilerOptions": {
+    "composite": true
+  },
+  "files": [
+    "./index.ts"
```

Comment:
For some reason, without `files` or `include`, `program.getProjectReferences()[0].commandLine.fileNames` was empty. I thought it should have resolved to all `*.ts` files by default?

Reply:
> To generate outputs for older TS versions, do I need to manually install each older version and run again?

No - we only run comparison tests for the latest TypeScript version. So 3 alone is fine

Reply:
When I was testing ad-hoc, the `reference.commandLine.fileNames` was working correctly, but I think I was using typescript@next... I'll have to investigate more.

Reply:
Ah, I was mistaken:

> All implementation files must be matched by an include pattern or listed in the files array. If this constraint is violated, tsc will inform you which files weren’t specified

https://www.typescriptlang.org/docs/handbook/project-references.html#composite

It seems it's working as intended, but via our programmatic route we don't get that warning. Maybe need to add our own.

Reply:
>  can the comparison test’s "patch" feature be used to patch the test's webpack.config.js itself?

No I'm afraid not; the tests rely upon an unchanged webpack config between patches. Sorry

Reply:
> It seems it's working as intended, but via our programmatic route we don't get that warning. Maybe need to add our own.

I'm not sure if this can be done reliably and efficiently... without actually doing a build of the dependent projects, it's hard to tell if a given file was intended to be included by one of them, or was just intended to be processed by the parent project (i.e. in our case, run through the loader). Implementing a warning from ts-loader side while _consuming_ project references might be out of scope—hopefully it's something users will notice while actually running `tsc` on those projects, since pre-building the referenced projects is currently required, and `tsc` will emit the error if `files` or `include` don't cover all the files.